### ISSUE TITLE
Daemon fails to rebuild file if original upload location not available

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry#1380]
## The Issue
If you publish a file, the daemon records the path to the published file. If this path is no longer valid, the daemon tries to rebuild it in the same location and fails. As a fallback/potential solution, it should rebuild it in the downloads directory. 

```
2018-08-21 18:26:56,999 INFO     lbrynet.daemon.Daemon:1375: Already have lbry file but missing file in C:\Users\thoma\Desktop\test delete, rebuilding it
2018-08-21 18:26:57,000 ERROR    lbrynet.lbry_file.client.EncryptedFileDownloader:158: Traceback (most recent call last):
  File "site-packages\lbrynet\lbry_file\client\EncryptedFileDownloader.py", line 155, in open_file
IOError: [Errno 2] No such file or directory: 'C:\\Users\\thoma\\Desktop\\test delete\\3702266720218818969.jpg'

2018-08-21 18:26:57,012 WARNING  lbrynet.daemon.auth.server:306: error processing api request: Failed to open C:\Users\thoma\Desktop\test delete\3702266720218818969.jpg. Make sure you have permission to save files to that location.
traceback:   File "site-packages\twisted\internet\defer.py", line 1297, in _inlineCallbacks
    
  File "site-packages\twisted\python\failure.py", line 389, in throwExceptionIntoGenerator
    
  File "site-packages\lbrynet\daemon\auth\server.py", line 163, in _inner
    
  File "site-packages\twisted\internet\defer.py", line 1297, in _inlineCallbacks
    
  File "site-packages\twisted\python\failure.py", line 389, in throwExceptionIntoGenerator
    
  File "site-packages\lbrynet\daemon\Daemon.py", line 1376, in jsonrpc_get
    
  File "site-packages\twisted\internet\defer.py", line 1297, in _inlineCallbacks
    
  File "site-packages\twisted\python\failure.py", line 389, in throwExceptionIntoGenerator
    
  File "site-packages\lbrynet\file_manager\EncryptedFileDownloader.py", line 131, in _start
    
  File "site-packages\twisted\python\threadpool.py", line 246, in inContext
    
  File "site-packages\twisted\python\threadpool.py", line 262, in <lambda>
    
  File "site-packages\twisted\python\context.py", line 118, in callWithContext
    
  File "site-packages\twisted\python\context.py", line 81, in callWithContext
    
  File "site-packages\lbrynet\lbry_file\client\EncryptedFileDownloader.py", line 161, in open_file
    
exceptions.ValueError: Failed to open C:\Users\thoma\Desktop\test delete\3702266720218818969.jpg. Make sure you have permission to save files to that location.
```

### Steps to reproduce
1. Publish a file from a folder that can be delete
2. Delete folder after publishing
3. Try to view the file in-app

## System Configuration

 

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

**Comment by tzarebczan** [comment]
My shot with https://github.com/lbryio/lbry/pull/1760 didn't work, it still tries to download to the original location (not sure where it's getting this from). The file_list does show the default download directory.

**Comment by tzarebczan** [comment]
This is now fixed in 0.31 :)
